it/test/connectors/ITSAStatusUpdateConnectorISpec.scala

```diff
@@ -0,0 +1,254 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import _root_.helpers.{ComponentSpecBase, WiremockHelper}
+import connectors.itsastatus.ITSAStatusUpdateConnectorModel.{ErrorItem, ITSAStatusUpdateResponseFailure, ITSAStatusUpdateResponseSuccess}
+import connectors.itsastatus.{ITSAStatusUpdateConnector, ITSAStatusUpdateRequest}
+import models.incomeSourceDetails.TaxYear
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.http.Status.{BAD_REQUEST, NO_CONTENT, OK}
+import play.api.libs.json.Json
+
+class ITSAStatusUpdateConnectorISpec extends AnyWordSpec with ComponentSpecBase {
+
+  lazy val connector: ITSAStatusUpdateConnector = app.injector.instanceOf[ITSAStatusUpdateConnector]
+
+  val optOutUpdateReason = "10"
+  val optInUpdateReason = "11"
+  val taxYear2024 = TaxYear.forYearEnd(2024)
+  val taxYear2023 = TaxYear.forYearEnd(2023)
+  val taxableEntityId = "AB123456A"
+
+  val correctOptOutRequestBody =
+    """{
+      | "taxYear":"2023-24",
+      | "updateReason":"10"
+      |}""".stripMargin
+
+  val correctOptIntRequestBody =
+    """{
+      | "taxYear":"2023-24",
+      | "updateReason":"11"
+      |}""".stripMargin
+
+  "ITSAStatusUpdateConnector" when {
+
+    ".updateITSAStatus()" when {
+
+      "happy path" should {
+
+        "return a successful response" in {
+
+          WiremockHelper.stubPut(s"/income-tax-view-change/itsa-status/update/$taxableEntityId", NO_CONTENT, "{}")
+
+          val requestBody = ITSAStatusUpdateRequest(taxYear = connector.toApiFormat(taxYear2024), updateReason = optOutUpdateReason)
+
+          val result = connector.updateITSAStatus(taxableEntityId, requestBody).futureValue
+
+          result.status shouldBe NO_CONTENT
+          result.body shouldBe ""
+
+          WiremockHelper.verifyPut(
+            uri = s"/income-tax-view-change/itsa-status/update/$taxableEntityId",
+            optRequestBody = Some(Json.stringify(Json.parse(correctOptOutRequestBody)))
+          )
+        }
+      }
+
+      "unhappy path" when {
+
+        "BAD_REQUEST" in {
+
+          val failureResponseBody =
+            """{
+              |  "failures": [
+              |    {
+              |      "code": "INVALID_PAYLOAD",
+              |      "reason": "Submission has not passed validation. Invalid payload."
+              |    }
+              |  ]
+              |}""".stripMargin
+
+          WiremockHelper.stubPut(s"/income-tax-view-change/itsa-status/update/$taxableEntityId", BAD_REQUEST, failureResponseBody)
+
+          val requestBody = ITSAStatusUpdateRequest(taxYear = connector.toApiFormat(taxYear2024), updateReason = optOutUpdateReason)
+
+          val result = connector.updateITSAStatus(taxableEntityId, requestBody).futureValue
```

Comment:
Would it be possible to test via the public `optIn()` and `optOut()` methods rather than exposing this internal method?

Reply:
duplicated the tests for opt in/opt out, changed the expected request bodies depending on opt in opt out.

Reply:
opt in opt out should use the underlying so the expected should be the tax year and updated reason code should match the respective journey type. 

feel so long as the model reads and writes are correct then we should be good. underlying methods are tested, opt in and opt out methods are now covered to update depending on the journey type.

for example now the tests fail for say an opt out journey update method if the stubs are expecting opt in request bodies in the put. The `withRequestBody()` I think is working

<img width="1438" alt="Screenshot 2024-10-01 at 11 15 07" src="https://github.com/user-attachments/assets/464c51ed-d0ed-48e0-8286-8b93624859c3">
<img width="1792" alt="Screenshot 2024-10-01 at 11 15 19" src="https://github.com/user-attachments/assets/b302d2aa-3b0a-467b-b823-972f3a330faa">